code/__DEFINES/icon_smoothing.dm

```diff
@@ -0,0 +1,124 @@
+/* smoothing_flags */
+/// Smoothing system in where adjacencies are calculated and used to build an image by mounting each corner at runtime.
+#define SMOOTH_CORNERS (1<<0)
+/// Smoothing system in where adjacencies are calculated and used to select a pre-baked icon_state, encoded by bitmasking.
+#define SMOOTH_BITMASK (1<<1)
+/// Atom has diagonal corners, with underlays under them.
+#define SMOOTH_DIAGONAL_CORNERS	(1<<2)
+/// Atom will smooth with the borders of the map.
+#define SMOOTH_BORDER	(1<<3)
+/// Atom is currently queued to smooth.
+#define SMOOTH_QUEUED	(1<<4)
+/// Smooths with objects, and will thus need to scan turfs for contents.
+#define SMOOTH_OBJ		(1<<5)
+
+
+DEFINE_BITFIELD(smoothing_flags, list(
+	"SMOOTH_CORNERS" = SMOOTH_CORNERS,
+	"SMOOTH_BITMASK" = SMOOTH_BITMASK,
+	"SMOOTH_DIAGONAL_CORNERS" = SMOOTH_DIAGONAL_CORNERS,
+	"SMOOTH_BORDER" = SMOOTH_BORDER,
+	"SMOOTH_QUEUED" = SMOOTH_QUEUED,
+	"SMOOTH_OBJ" = SMOOTH_OBJ,
+))
+
+
+/*smoothing macros*/
+
+#define QUEUE_SMOOTH(thing_to_queue) if(thing_to_queue.smoothing_flags & (SMOOTH_CORNERS|SMOOTH_BITMASK)) {SSicon_smooth.add_to_queue(thing_to_queue)}
+
+#define QUEUE_SMOOTH_NEIGHBORS(thing_to_queue) for(var/atom/atom_neighbor as anything in orange(1, thing_to_queue)) {QUEUE_SMOOTH(atom_neighbor)}
+
+/**SMOOTHING GROUPS
+ * Groups of things to smooth with.
+ * * Contained in the `list/smoothing_groups` variable.
+ * * Matched with the `list/canSmoothWith` variable to check whether smoothing is possible or not.
+ */
+#define S_TURF(num) ((24 * 0) + num) //Not any different from the number itself, but kept this way in case someone wants to expand it by adding stuff before it.
```

Comment:
Do we really need to have `24*0` before it? Having the define as just num is good enough, since it allows us to add in negative sorting groups if we wanted for some reason by just adding the number offset before.